src/timebar/components/date-selector.js

```diff
@@ -1,32 +1,97 @@
 import React, { Component } from 'react'
 import PropTypes from 'prop-types'
+import cx from 'classnames'
 import styles from './date-selector.module.css'
 import { ReactComponent as IconArrowUp } from '../icons/arrowUp.svg'
 import { ReactComponent as IconArrowDown } from '../icons/arrowDown.svg'
 
 class DateSelector extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      value: props.value,
+      error: false,
+    }
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (this.props.value !== nextProps.value) {
+      this.setState({ value: nextProps.value })
+    }
+  }
+
+  validate(value) {
+    const { unit } = this.props
+    const isNumber = typeof value === 'number'
+    if (!isNumber) return false
+
+    // TODO: validate days / months / years
```

Comment:
I think you could get away with just using https://github.com/iamkun/dayjs/blob/dev/docs/en/API-reference.md#validation-isvalid
You'd need to move the validation step to onStartChange/onEndChange in TimerangeSelector

Reply:
Good catch, now it has a better validation as also check if the day is included in the current month selected